Packs/Inventa/Integrations/Inventa/Inventa.py

```diff
@@ -0,0 +1,465 @@
+
+
+def get_dsar_piis_command(client: Client, ticket_id: int) -> CommandResults:
+    dsar = client.get_dsar(ticket_id)
+    piis = dsar.get("piis", [])
+    pii_list = list()
+    for pii in piis:
+        pii_list.append(pii.get("piiEntityType", ""))
+
+    demisto.debug(f"{piis}")
+    demisto.debug(f"{pii_list}")
+
+    return CommandResults(outputs={"piis": list(set(pii_list))},
+                          outputs_prefix="Inventa.Dsar.Piis")
+
+
+def get_dsar_transactions_command(client: Client, ticket_id: int) -> CommandResults:
+    dsar = client.get_dsar(ticket_id)
+    transactions = dsar.get("copiesUsageData", {}).get("transactions", [])
+    if "entityTypes" in transactions:
+        transactions["entityTypes"] = [transactions["entityTypes"]["type"]
+                                       for item in transactions["entityTypes"] if item == "type"]
+
+    for transaction in transactions:
+        demisto.debug(f"file: {transaction}")
+        if "entityTypes" in transaction:
+            entityTypes = transaction["entityTypes"]
+            demisto.debug(f"types: {entityTypes}")
+            stripped = [item["type"] for item in entityTypes]
+            demisto.debug(f"stripped: {stripped}")
+            transaction["entityTypes"] = stripped
+    if transactions:
+        return CommandResults(outputs={"transactions": transactions},
+                              outputs_prefix="Inventa.Dsar.Transactions",
+                              outputs_key_field="id")
+    else:
+        return CommandResults(outputs={"transactions": "None"},
+                              outputs_prefix="Inventa.Dsar.Transactions",
+                              outputs_key_field="id")
+
+
+def get_dsar_files_command(client: Client, ticket_id: int) -> CommandResults:
+    dsar = client.get_dsar(ticket_id)
+    files = dsar.get("copiesUsageData", {}).get("files", [])
+    demisto.debug(f"{files}")
+    for file in files:
+        # file.pop("path")
+        # file.pop("url")
+        demisto.debug(f"file: {file}")
+        if "entityTypes" in file:
+            entityTypes = file["entityTypes"]
+            demisto.debug(f"types: {entityTypes}")
+            stripped = [item["type"] for item in entityTypes]
+            stripped = list(set(stripped))
+            demisto.debug(f"types: {stripped}")
+            file["entityTypes"] = ", ".join(stripped)
+        if "timestamp" in file:
+            ts = file.get("timestamp", 0)
+            if type(ts) is int:
+                file["timestamp"] = datetime.utcfromtimestamp(
+                    float(f"{str(ts)[:10]}.{str(ts)[10:]}")).strftime("%d %b %Y %H:%M:%S")
+            # file["entityTypes"] = stripped
+    if files:
+        return CommandResults(outputs={"files": files},
+                              outputs_prefix="Inventa.Dsar.Files",
+                              outputs_key_field="id")
+
+    else:
+        return CommandResults(outputs={"files": "None"},
```

Comment:
"None" as a string?

Reply:
Currently it's a placeholder value. Next update will slightly revise the API call and the return value if no files (or other storage types) are available for this datasubject